app/controllers/badges_controller.rb

```diff
@@ -1,6 +1,7 @@
 class BadgesController < ApplicationController
   before_action :authenticate_user!
   before_action :set_badge, only: [:show]
+  load_and_authorize_resource except: [:require, :redeem]
```

Comment:
@frmendes After messing around with the abilities for awhile, the only one that seemed to work was `can :manage, Badge`, of course I am not gonna given authorisation to every user to manage Badges just so they can redeem badges. Do you have any idea how to fix this? ...